src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarily.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+{
+    /// <summary>CA1805: Do not initialize unnecessarily.</summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotInitializeUnnecessarilyAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1805";
+
+        internal static DiagnosticDescriptor DefaultRule = DiagnosticDescriptorHelper.Create(RuleId,
+            new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.DoNotInitializeUnnecessarilyTitle), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources)),
+            new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.DoNotInitializeUnnecessarilyMessage), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.DoNotInitializeUnnecessarilyDescription), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources)),
+            isPortedFxCopRule: true,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            analysisContext.RegisterOperationAction(context =>
+            {
+                var init = (IFieldInitializerOperation)context.Operation;
+
+                foreach (var field in init.InitializedFields)
+                {
+                    // If a field uses a value known to be its default, warn.
+                    if (!field.IsConst && init.Value != null && UsesKnownDefaultValue(init, field))
+                    {
+                        context.ReportDiagnostic(init.CreateDiagnostic(DefaultRule, field.Name));
```

Comment:
Will this generate duplicate diagnostics for VB where an initializer might apply to multiple fields? https://github.com/dotnet/roslyn/blob/730e63429a6691f076f75556a61ef759b0d96743/src/Compilers/VisualBasic/Test/IOperation/IOperation/IOperationTests_ISymbolInitializer.vb#L175-L188

I am wondering if we should just process the first field symbol as the modifiers and initializers would be shared by all fields for such a declaration.

Reply:
> I am wondering if we should just process the first field symbol

Ok